File: Neetcode Questions/Two Pointer/container-with-most-water.cpp

// https://leetcode.com/problems/container-with-most-water/

class Solution {
public:
    int maxArea(vector<int>& height) {
        
        int maxCapacity = 0;
        int n = height.size();
        if(n == 0) return maxCapacity;

        int left = 0, right = n-1;
        while(left < right){
            int currentCapacity = (right - left)*min(height[left], height[right]);
            maxCapacity = max(maxCapacity, currentCapacity);
            height[left] <= height[right] ? left++ : right--;
        }

        return maxCapacity;
    }
};

File: Neetcode-150-js/Two Pointers/container-with-most-water.js

// https://leetcode.com/problems/container-with-most-water/description/

/**
 * @param {number[]} height
 * @return {number}
 */
var maxArea = function(height) {
    let ans = 0;
    let len = height.length;
    if(len === 0) return ans;

    let i = 0, j = len-1;
    while(i < j){
        let smallAns = (j-i)*Math.min(height[i], height[j]);
        ans = Math.max(ans, smallAns);
        (height[i] <= height[j]) ? i++ : j--;
    }
    return ans;
};